sympy/simplify/fu.py

```diff
@@ -1020,6 +1020,48 @@ def f(rv):
 
     return bottom_up(rv, f)
 
+def _TR11(rv):
+    """
+    Helper for TR11 to find the `base` argument of TR11.
+
+    Examples
+    ========
+
+    >>> from sympy.simplify.fu import _TR11 as T
+    >>> from sympy import cos, sin, pi
+    >>> from sympy.abc import x
+    >>> T(sin(x/3)*sin(2*x)*sin(x/4)/(cos(x/6)*cos(x/8)))
+    4*sin(x/8)*sin(x/6)*sin(2*x)
+    >>> T(sin(x/3)/cos(x/6))
+    2*sin(x/6)
+    >>> T(sin(2*x)*cos(x/8)/sin(x/4))
+    sin(2*x)/(2*sin(x/8))
+
+    """
+    def f(rv):
+        arg_denoms = dict()
+        for fi in rv.args:
+            if fi.is_Pow and fi.exp.is_negative and fi.base.func in (sin, cos):
+                arg_denoms[fi.base.args[0]] = fi.base.func
+
+            elif fi.func in (sin, cos):
+                if len(arg_denoms):
+                    arg_num = fi.args[0]
+                    for arg_denom in list(arg_denoms):
```

Comment:
```suggestion
                    for arg_denom in ordered(arg_denoms):
```

This will be more deterministic if you make the items ordered. If there are two possible matches then we should always pick the same one first. I am not sure if this should also be reversed, but you can test. e.g. `sin(4*x)*sin(6*x)/sin(x)`.

Reply:
> This will be more deterministic if you make the items ordered.

Thanks for the suggestion. I'll update it accordingly.

Reply:
maybe like this?
```diff
diff --git a/sympy/simplify/fu.py b/sympy/simplify/fu.py
index fb9f5eb..3fe6ffb 100644
--- a/sympy/simplify/fu.py
+++ b/sympy/simplify/fu.py
@@ -1039,24 +1039,23 @@ def _TR11(rv):

     """
     def f(rv):
-        arg_denoms = dict()
+        arg_denoms = defaultdict(set)
         for fi in rv.args:
             if fi.is_Pow and fi.exp.is_negative and fi.base.func in (sin, cos):
-                arg_denoms[fi.base.args[0]] = fi.base.func
-
-            elif fi.func in (sin, cos):
-                if len(arg_denoms):
-                    arg_num = fi.args[0]
-                    for arg_denom in list(arg_denoms):
-                        base = arg_num/arg_denom
-                        if not (fi.func == cos and arg_denoms[arg_denom] == cos) :
-                            if base % 2 == 0:
-                                rv = TR11(rv, arg_denom)
-                                arg_denoms.pop(arg_denom)
-
-                            elif (1/base) % 2 == 0:
-                                rv = TR11(rv, arg_num)
-                                arg_denoms.pop(arg_denom)
+                arg_denoms[fi.base.args[0]].add(fi.base.func)
+
+            elif fi.func in (cos, sin):
+                arg_num = fi.args[0]
+                for arg_denom in ordered(arg_denoms):
+                    base = arg_num/arg_denom
+                    if not (fi.func == cos and sin not in arg_denoms[arg_denom]
):
+                        if base % 2 == 0:
+                            rv = TR11(rv, arg_denom)
+                            arg_denoms.pop(arg_denom)
+
+                        elif (1/base) % 2 == 0:
+                            rv = TR11(rv, arg_num)
+                            arg_denoms.pop(arg_denom)

         return rv
```

Reply:
> maybe like this?

@smichr , Sorry for replying late. Thanks for the diff. I will update the code this weekend after my mid semester exams will be over. I apologise for the delay.

Reply:
no problem